src/rest.ts

```diff
@@ -0,0 +1,88 @@
+import http = require('http')
+import net = require('net')
+import { CompanionSatelliteClient } from './client'
+
+export class RestServer {
+    private _cs_client: CompanionSatelliteClient
+
+    constructor(client: CompanionSatelliteClient, port: Number) {
+
+        this._cs_client = client
+        // super()
+
+        const server = http.createServer(async (req, res) => {
+            //set the request route
+            if (req.method === 'GET') {
+                console.log("api get: ", req.url)
+                switch (req.url) {
+                    //#TODO:add host case
+                    case ('/api/ip'): {
```

Comment:
Do these need to be separate? I think it would be nicer to have a single endpoint (`/api/host`, `/api/address`, or something, maybe even `/api/config`) that returns json containing both the host and port it is configured to use

Reply:
it wold be nice to do some type checking on the ip addrese to make sure it is valid, but that only works if we know the user intended it to be an ip and not a host address.

maybe that is to much ?